holocron/trainer/core.py

```diff
@@ -366,6 +366,62 @@ def _eval_metrics_str(eval_metrics: Dict[str, float]) -> str:
                 f"(Acc@1: {eval_metrics['acc1']:.2%}, Acc@5: {eval_metrics['acc5']:.2%})")
 
 
+class BinaryClassificationTrainer(Trainer):
+    """Image binary classification trainer class
+
+    Args:
+        model (torch.nn.Module): model to train
+        train_loader (torch.utils.data.DataLoader): training loader
+        val_loader (torch.utils.data.DataLoader): validation loader
+        criterion (torch.nn.Module): loss criterion
+        optimizer (torch.optim.Optimizer): parameter optimizer
+        gpu (int, optional): index of the GPU to use
+        output_file (str, optional): path where checkpoints will be saved
+        use_sigmoid (bool, optional): apply sigmoid function to the output
+    """
+
+    def __init__(self, model, train_loader, val_loader, criterion, optimizer, gpu=None, output_file='./checkpoint.pth',
```

Comment:
If we manage to find a way to avoid modifying the constructor, that would be even better. Perhaps setting the attribute by default (not in the constructor)?

If not, we need to add some proper typing here

Reply:
actually for now, this would be simpler to use the torch.sigmoid by default in the eval run. I'll think about how to handle this non-sigmoid option later on